### PR TITLE
vscode: add support for inline completions

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Closes: #11868.

The pull-request adds support for the VS Code `InlineCompletions` related APIs (non-proposed):
- [`InlineCompletionItem`](https://code.visualstudio.com/api/references/vscode-api#InlineCompletionItem)
- [`InlineCompletionItemProvider`](https://code.visualstudio.com/api/references/vscode-api#InlineCompletionItemProvider)
- [`InlineCompletionList`](https://code.visualstudio.com/api/references/vscode-api#InlineCompletionList)
- [`InlineCompletionTriggerKind`](https://code.visualstudio.com/api/references/vscode-api#InlineCompletionTriggerKind)
- [`InlineCompletionContext`](https://code.visualstudio.com/api/references/vscode-api#InlineCompletionContext)
- `registerInlineCompletionItemProvider`

https://user-images.githubusercontent.com/40359487/203127679-b7e19ebb-3115-4add-916f-1701a4f06a36.mp4

#### How to test

1. start the application with the following https://github.com/eclipse-theia/theia/files/10059321/inline-completions-0.0.1.zip
2. open an editor
3. `hello world` should be proposed as an inline completion

```ts

import * as vscode from 'vscode';

export function activate(context: vscode.ExtensionContext) {
    const provider = createInlineCompletionItemProvider();
    vscode.languages.registerInlineCompletionItemProvider({ pattern: '**' }, provider);
}

export function deactivate() { }

function createInlineCompletionItemProvider(): vscode.InlineCompletionItemProvider {
    return {
        provideInlineCompletionItems(document, position, context, token) {
            return [{
                insertText: "hello world",
            }];
        },
    }
}
```

#### CQ

- [x] [CQ Approved](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/5022)

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
The CQ for the pull-request has been approved :+1:

File: packages/plugin-ext/src/plugin/types-impl.ts [hunk 1]
```diff
@@ -992,6 +992,37 @@ export class CompletionList {
     }
 }
 
+export enum InlineCompletionTriggerKind {
+    Invoke = 0,
+    Automatic = 1,
+}
+
+@es5ClassCompat
+export class InlineSuggestion implements theia.InlineCompletionItem {
```
Comment (msujew) [inline packages/plugin-ext/src/plugin/types-impl.ts:1001]: Question: Is there a specific reason to deviate from the original `InlineCompletionItem` name for this class implementation? Same question for the `InlineSuggestionList` below.

**Review by msujew — COMMENTED** [review]
Looks quite good to me. I can confirm that the functionality works as expected with the provided extension and the APIs show as supported in the comparator report.

I have a question regarding naming, see below. Otherwise LGTM 👍

File: packages/plugin-ext/src/plugin/types-impl.ts (hunk 1 above)
Comment (vince-fugnitto) [reply packages/plugin-ext/src/plugin/types-impl.ts:1001]: @msujew I agree, I wasn't too fond of having the different names here as well but it came straight from VS Code. I don't think we need to follow in this case.

Comment (msujew) [reply packages/plugin-ext/src/plugin/types-impl.ts:1001]: I see, interesting. Yeah, we can probably get away with changing the class name to `InlineCompletionItem`.

File: packages/plugin-ext/src/plugin/plugin-context.ts [hunk 2]
```diff
@@ -1124,7 +1130,10 @@ export function createAPIFactory(
             TestTag,
             TestRunRequest,
             TestMessage,
-            ExtensionKind
+            ExtensionKind,
+            InlineCompletionItem: InlineCompletionItem,
+            InlineCompletionList: InlineCompletionList,
```
Comment (msujew) [inline packages/plugin-ext/src/plugin/plugin-context.ts:1135]: Minor:
```suggestion
            InlineCompletionItem,
            InlineCompletionList,
```

**Review by msujew — APPROVED** [review]
LGTM 👍

File: packages/plugin-ext/src/plugin/plugin-context.ts (hunk 2 above)
Comment (vince-fugnitto) [reply packages/plugin-ext/src/plugin/plugin-context.ts:1135]: Thanks, the result of just doing a rename :)

**Pull request merged**